dev-client/src/components/inputs/Select.tsx

```diff
@@ -0,0 +1,162 @@
+import {TextInput} from 'react-native-paper';
+import {theme} from 'terraso-mobile-client/theme';
+
+// utility type so we can strictly validate the types of inputs/callbacks
+// based on whether the select is nullable
+type NullableIf<T, Nullable extends boolean> = Nullable extends true
+  ? T | null
+  : T;
+
+export type SelectProps<T, Nullable extends boolean> = {
+  nullable: Nullable;
+  options: T[] | readonly T[];
+  renderValue: (_: T) => string;
+  value: NullableIf<T, Nullable>;
+  onValueChange: (value: NullableIf<T, Nullable>) => void;
+  label?: string;
+  unselectedLabel?: string;
+  disabled?: boolean;
+} & ViewStyle;
+
+export const Select = <T, Nullable extends boolean>({
+  nullable,
+  options,
+  onValueChange,
+  renderValue,
+  value,
+  label,
+  unselectedLabel,
+  disabled = false,
+  ...style
+}: SelectProps<T, Nullable>) => {
+  const {t} = useTranslation();
+  const ref = useRef<ModalHandle>(null);
+  const onClose = useCallback(() => ref.current?.onClose(), [ref]);
+
+  const renderOption = useCallback(
+    ({item: option}: {item: T | null}) => (
+      <Pressable
+        accessibilityState={{
+          selected: value === option,
+        }}
+        onPress={() => {
+          onValueChange(option as T);
+          onClose();
+        }}>
+        <Row
+          backgroundColor={
+            option === value ? 'input.filled.enabledFill' : 'primary.contrast'
+          }
+          justifyContent="flex-start"
+          alignItems="center"
+          paddingVertical="sm"
+          paddingHorizontal="sm">
+          <Icon
+            color={option === value ? 'action.active' : 'primary.contrast'}
+            name="check"
+            size="sm"
+          />
+          <Box width="sm" />
+          <Text variant="body1" color="text.primary">
+            {option ? renderValue(option) : unselectedLabel}
+          </Text>
+        </Row>
+      </Pressable>
+    ),
+    [value, renderValue, onValueChange, unselectedLabel, onClose],
+  );
+
+  const Header = useMemo(
+    () => (
+      <Pressable onPress={onClose}>
+        <Row
+          backgroundColor="primary.main"
+          justifyContent="flex-end"
+          alignItems="center"
+          padding="md">
+          <Text
+            variant="body1-strong"
+            color="primary.contrast"
+            textTransform="uppercase">
+            {t('general.done')}
+          </Text>
+        </Row>
+      </Pressable>
+    ),
+    [t, onClose],
+  );
+
+  const optionsWithNull = useMemo(
+    () => (nullable ? [null, ...options] : options),
+    [nullable, options],
+  );
+
+  return (
+    <BottomSheetModal
+      ref={ref}
+      scrollable={false}
+      CloseButton={null}
+      trigger={onOpen => (
+        <Pressable
+          accessibilityState={{disabled}}
+          onPress={disabled ? null : onOpen}>
+          <TextInput
+            editable={false}
+            value={value === null ? '' : renderValue(value)}
```

Comment:
is this equivalent?
```suggestion
            value={value ? renderValue(value) : null}
```

Reply:
they aren't equivalent no, the type of value is generic so if its type was number and the value was 0 (which is falsy) then `value ? renderValue(value) : null` would return `null` when we want it to return `renderValue(0)`